src/main/java/task/Task.java

```diff
@@ -0,0 +1,71 @@
+package task;
+
+import java.io.Serializable;
+
+/**
+ * Class encapsulating a single task to be done (by the user) in Kniaz
+ */
+public abstract class Task implements Serializable {
+
+    /**
+     * Whether this task is done
+     */
+    private boolean isDone = false;
+
+
+    /**
+     * Name of this task
+     */
+    private String taskName = "";
+
+    /**
+     * The (protected) constructor for this task, mostly for inheritance by subclasses
+     * @param taskName name of task
+     * @param isDone whether it is done
+     */
+    protected Task(String taskName, boolean isDone) {
+        this.taskName = taskName;
+        this.isDone = isDone;
+    }
+
+
+
+    /**
+     * Marks this task as done, does NOT check for if it was already done
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks this task as not done, does NOT check for if it was already undone
+     */
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+    /**
+     * The user-facing string representation of this Task, containing information about
+     * whether this task is done and its name
+     * @return the user-facing string representation of this ToDo.
+     */
+    public String toPrintString() {
+        String statusIcon = "";
+        if (this.isDone) {
+            statusIcon = "X";
+        } else if (!this.isDone) {
+            statusIcon = " ";
+        }
+        return String.format("[%s] %s", statusIcon, this.taskName);
+        // return in format [statusIcon] taskname
+    }
```

Comment:
1 small nitpick is that the output can be dictated by modifying the object toString method, and then you can just return the object itself. But otherwise this is also alright!